cmake/Modules/ArrowExternalProject.cmake

```diff
@@ -0,0 +1,74 @@
+# arrow external project
+# target:
+#  - arrow_ep
+# depends:
+#
+# this module defines:
+#  - ARROW_HOME
+#  - ARROW_SOURCE_DIR
+#  - ARROW_INCLUDE_DIR
+#  - ARROW_SHARED_LIB
+#  - ARROW_STATIC_LIB
+#  - PLASMA_INCLUDE_DIR
+#  - PLASMA_STATIC_LIB
+#  - PLASMA_SHARED_LIB
+
+option(BUILD_ARROW_PLASMA_JAVA_CLIENT
+    "build arrow plasma java client"
+    OFF)
+
+if (RAY_BUILD_JAVA)
+  set(BUILD_ARROW_PLASMA_JAVA_CLIENT ON)
+endif ()
+
+set(arrow_URL https://github.com/apache/arrow.git)
+set(arrow_TAG 4660833b2c5ef63a97445e304b8f72a2e0170f9c)
```

Comment:
In current Arrow building process, if a user pulls a commit with the commit id changed, the script will sense the mismatch and will trigger arrow rebuild. Is this also supported? Otherwise, pulling code may cause building failure and users don't why.

Reply:
here I just catch up with the script `build_arrow.sh`... it use commit id.

cmake will just update the arrow source code and recompile arrow libraries. cmake externalproject module has good update mechanism to control it. which is much better then the script.